src/fsharp/FSharp.Core/control.fs

```diff
@@ -846,6 +847,14 @@ namespace Microsoft.FSharp.Control
         // delayPrim = "bindA (return ()) f"
         let delayA f = callA f ()
 
+        let protectExn (cexn:OperationCanceledException) (edi:ExceptionDispatchInfo) =
+            let exn = edi.GetAssociatedSourceException()
+            let collected =
+                match cexn.InnerException with
+                | null -> [|exn|]
```

Comment:
Why are we throwing an AggregateException as the inner exception when there's only one exception? Feels a bit superfluous.